examples/ClientGrants.java

```diff
@@ -0,0 +1,135 @@
+
+  private static final String POLICY =
+      new StringBuilder()
+          .append("{\n")
+          .append("    \"Statement\": [\n")
+          .append("       " + " {\n")
+          .append("            \"Action\": [\n")
+          .append("                \"s3:GetBucketLocation\",\n")
+          .append("                \"s3:ListBucket\"\n")
+          .append("            ],\n")
+          .append("            \"Effect\": \"Allow\",\n")
+          .append("            \"Principal\": \"*\",\n")
+          .append("            \"Resource\": \"arn:aws:s3:::test\"\n")
+          .append("        }\n")
+          .append("    ],\n")
+          .append("    \"Version\": \"2012-10-17\"\n")
+          .append("}\n")
+          .toString();
+
+  static class JwtToken {
+
+    @JsonProperty("access_token")
+    private final String accessToken;
+
+    @JsonProperty("expires_in")
+    private final long expiredAfter;
+
+    @ConstructorProperties({"access_token", "expires_in"})
+    public JwtToken(String accessToken, long expiredAfter) {
+      this.accessToken = accessToken;
+      this.expiredAfter = expiredAfter;
+    }
+  }
+
+  @SuppressWarnings({"SameParameterValue", "squid:S1192"})
+  static ClientGrantsToken getTokenAndExpiry(
+      @Nonnull String clientId,
+      @Nonnull String clientSecret,
+      @Nonnull String idpClientId,
+      @Nonnull String idpEndpoint) {
+    Objects.requireNonNull(clientId, "Client id must not be null");
+    Objects.requireNonNull(clientSecret, "ClientSecret must not be null");
+
+    final RequestBody requestBody =
+        new FormBody.Builder()
+            .add("username", clientId)
+            .add("password", clientSecret)
+            .add("grant_type", "password")
+            .add("client_id", idpClientId)
+            .build();
+
+    final Request request = new Request.Builder().url(idpEndpoint).post(requestBody).build();
+
+    final OkHttpClient client = new OkHttpClient();
+    try (Response response = client.newCall(request).execute()) {
+      final ObjectMapper mapper = new ObjectMapper();
+      mapper.disable(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES);
+      mapper.setVisibility(
+          VisibilityChecker.Std.defaultInstance()
+              .withFieldVisibility(JsonAutoDetect.Visibility.ANY));
+
+      final JwtToken jwtToken =
+          mapper.readValue(Objects.requireNonNull(response.body()).charStream(), JwtToken.class);
+      return new ClientGrantsToken(jwtToken.accessToken, jwtToken.expiredAfter, POLICY);
+    } catch (IOException e) {
+      throw new IllegalStateException(e);
+    }
+  }
+
+  @SuppressWarnings("squid:S106")
+  public static void main(String[] args) throws Exception {
+    final String clientId = "user";
+    final String clientSecret = "password";
+    final String idpEndpoint =
+        "http://idp-host:idp-port/auth/realms/master/protocol/openid-connect/token";
+    final String stsEndpoint = "http://sts-host:sts-port/sts";
```

Comment:
```suggestion
    // STS endpoint usually points to MinIO endpoint in case of MinIO
    final String stsEndpoint = "http://sts-host:sts-port/";
```